packages/xstate-wallet/src/channel-wallet.ts

```diff
@@ -286,12 +303,39 @@ export class ChannelWallet {
     }
   }
 
-  approveRichObjective(channelId: string) {
+  public async approveRichObjective(channelId: string): Promise<void> {
     const richObjective = this.store.richObjectives[channelId];
     if (!richObjective) {
       throw new Error(`Rich objective must exist for channelId ${channelId}`);
     }
-    this.crankRichObjectives({type: 'Approval'});
+
+    await this.crankRichObjectives({type: 'Approval'});
+
+    // TODO: need a better way to figure out when to broadcast an objective
+    //        since we don't know here if we are the creator of the objective or received the objective from elsewhere
+    if (!richObjective.myIndex) {
```

Comment:
```suggestion
    const amJoiner = richObjective.myIndex === 1;
    if (amJoiner) {
```

Reply:
Good call, but your logic above is the inverse of what we need. e01de9914

Reply:
All the more reason 😆